crates/runtime/tests/acceleration/on_conflict.rs

```diff
@@ -0,0 +1,227 @@
+        refresh_mode: Some(RefreshMode::Append),
+        refresh_check_interval: Some("1s".to_string()),
+        primary_key: Some("event_id".to_string()),
+        on_conflict: drop_hashmap,
+        ..Acceleration::default()
+    });
+    on_conflict_drop.time_column = Some("event_timestamp".to_string());
+
+    let df = get_test_datafusion();
+
+    let app = AppBuilder::new("on_conflict_behavior")
+        .with_dataset(on_conflict_upsert)
+        .with_dataset(on_conflict_drop)
+        .build();
+
+    let rt = Runtime::builder()
+        .with_app(app)
+        .with_datafusion(df)
+        .build()
+        .await;
+
+    // Set a timeout for the test
+    tokio::select! {
+        () = tokio::time::sleep(std::time::Duration::from_secs(10)) => {
+            return Err(anyhow::anyhow!("Timed out waiting for datasets to load"));
+        }
+        () = rt.load_components() => {}
+    }
+
+    // Wait for both dataset to be setup
+    assert!(
+        wait_until_true(Duration::from_secs(30), || async {
+            let mut query_result = rt
+                .datafusion()
+                .query_builder(
+                    "SELECT * FROM event_logs_upsert LIMIT 1",
+                    Protocol::Internal,
+                )
+                .build()
+                .run()
+                .await
+                .expect("result returned");
+            let mut batches = vec![];
+            while let Some(batch) = query_result.data.next().await {
+                batches.push(batch.expect("batch"));
+            }
+            !batches.is_empty() && batches[0].num_rows() == 1
+        })
+        .await,
+        "Expected 1 rows returned"
+    );
+    assert!(
+        wait_until_true(Duration::from_secs(30), || async {
+            let mut query_result = rt
+                .datafusion()
+                .query_builder("SELECT * FROM event_logs_drop LIMIT 1", Protocol::Internal)
+                .build()
+                .run()
+                .await
+                .expect("result returned");
+            let mut batches = vec![];
+            while let Some(batch) = query_result.data.next().await {
+                batches.push(batch.expect("batch"));
+            }
+            !batches.is_empty() && batches[0].num_rows() == 1
+        })
+        .await,
+        "Expected 1 rows returned"
+    );
+
+    let original_data = rt
+        .datafusion()
+        .ctx
+        .sql("SELECT * FROM event_logs_upsert")
+        .await?
+        .collect()
+        .await?;
+
+    db_conn
+        .conn
+        .execute(
+            "
+UPDATE event_logs
+SET event_name = 'File Accessed',
+    event_timestamp = '2024-08-24 15:45:00'
+WHERE event_name = 'File Download'
+  AND event_timestamp = '2023-05-17 13:20:00';
+",
+            &[],
+        )
+        .await
+        .expect("inserted data");
+
+    // Wait for result to be refreshed
+    tokio::time::sleep(std::time::Duration::from_secs(2)).await;
```

Comment:
Despite refresh is configured every 1s I would slightly increase timeout / 4-5s due to potential connectivity/transition/processing delay that can make this test flaky.